proposals/20200506-artifacts-scope-part-2.md

```diff
@@ -0,0 +1,135 @@
+# Falco Artifacts Scope - Part 2
+
+The **Falco Artifact Scope** proposal is divided in two parts:
+1. the [Part 1](./20200506-artifacts-scope-part-1.md): to “capture” the state of recent changes
+2. the Part 2 - *this document*: the intended state moving forward
+
+## Summary 
+
+See [Part 1](./20200506-artifacts-scope-part-1.md).
+
+## Terms
+
+See [Part 1](./20200506-artifacts-scope-part-1.md).
+
+## Packages
+
+Official packages for x86 64bits only.
+
+The following convention MUST be used for all packages.
+
+_All package names MUST contain a version._
+
+_If a package installs the Falco kernel module it MUST contain `module`._
+
+_If a package installs the Falco BPF probe it MUST contain `bpf`._
+
+_In general, if a package install a Falco driver it MUST contain the driver name._
+
+
+### .deb
+
+ Falco running in debian like systems that will default to the kernel module.
+
+- falco-*x.y.z*-x86.deb
+     - `falco` only (default depends on `falco-module`)
+ - falco-*x.y.z*-module-x86.deb
+     - `module` only
+ - falco-*x.y.z*-bpf-x86.deb
+     - `bpf` only
+ - falco-*x.y.z*-ptrace-x86.deb
+     - `ptrace` only
+
+N.B.:
+
+### .rpm
+
+ Falco running in rpm like systems that will default to the kernel module.
+
+- falco-*x.y.z*-x86.deb
+     - `falco` only (default depends on `falco-module`)
+- falco-*x.y.z*-module-x86.deb
+    - `module` only
+- falco-*x.y.z*-bpf-x86.deb
+    - `bpf` only
+- falco-*x.y.z*-ptrace-x86.deb
+    - `ptrace` only
+
+
+### .tar.gz
+
+- falco-bin-x86.tar.gz
+    - `falco` binary, `falco-loader-script`, and related dependecies
+    - `INSTALL` file
+    - `Makefile` file
+- falco-src-x86.tar.gz
+    - No binaries
+    - `INSTALL` file
+- falco-module-src-x86.tar.gz
+    - `module` sources with `Makefile`
+    - `INSTALL` file
+- falco-bpf-src-x86.tar.gz
+    - `bpf` sources with `Makefile`
+    - `INSTALL` file
+- falco-ptrace-src-x86.tar.gz
+    - `ptrace` sources with `Makefile`
+    - `INSTALL` file
+
+## Images
+
+The following convention MUST be used for all container images.
+
+*TODO* image name conventions
+
+ - falcosecurity/falco:TAG
+     - Runs `falco` userspace only
+ - falcosecurity/falco-driver-loader:TAG
+     - Runs `falco-driver-loader` and exit
+ - falcosecurity/*TBD**
```

Comment:
We could also think to name what is currently called `falcosecurity/falco:latest-slim` to something different to express the fact that this image will NOT try to build/download/install any driver, and it will require the driver to be already inserted in the host system.

So, naming game starts!

- falcosecurity/falco-slim
- falcosecurity/falco-userspace
- falcosecurity/falchetto

Reply:
Honestly I can't stand the word `slim` being used for any reason other than denote something taking up less persistent storage space.

My thought is to always name something very verbose and very accurate FIRST.

Then you can alias it later.

```bash
falcosecurity/falco-userspace-only:latest    # Only the userspace component of Falco, no drivers
falcosecurity/falco-slim                                # Alias of :point_up: 
```

Reply:
The same goes for `minimal`

I still don't really understand the difference between `slim` and `minimal` at a glance. 

I would wrather see names like this

```
falco-debian-toolchain-module-build:latest
falco-scratch-no-toolchain-bpf-downloader:latest
falco-debian-userspace-only:latest
falco-alpine-module-only:latest
falco-alpine-download-else-compile-module:latest
```

So we know exactly what they do, and then alias them to friendly names as needed.

Reply:
Although I like clarity and accuracy I still really think the name should NOT represent "what a thing does" but "what a thing is". 

If you think about `falcosecurity/falco:latest` (715MB) vs. `falcosecurity/falco:latest-slim` (229MB) the naming is actually correct, though not accurate because the behavior is different.  

That being said, I don't have a definitive answer to that, sorry :(

PS
BTW we're going to remove the `minimal`. See [here](https://github.com/falcosecurity/falco/pull/1184/files#diff-1f98b8cfa00350edfbedd86ce219e57cR100).

Reply:
I disagree @leogr 

Accessibility and User Experience are our top priority. 

The name should represent "what a thing does" first. Administrators and operators will be the folks using this. We need to cater to them.  

```bash
docker run falcosecurity/falco:latest # 99% of people will run this. 
docker run falcosecurity/falco-slim:latest # Those 99% mentioned above have no idea what the difference is between this and `falco`
```

We have to be generous to our users. We have to make adopting Falco successful. Right now it might make sense in your mind that `slim` or `minimal` has different expectations.

A new Falco user will never understand `falco-slim` vs `falco` without more detail. Most users will not research the bowels of github to find this detail.

Reply:
As I said I don't really agree with that, but I will not insist, it's just my opinion :hugs: 


So I just recap the issues here, then we can decide what to do.

**Issues to take into account**
- changing the current `falcosecurity/falco:lastest` will break installations
- an image can have several behaviors and can run with different options (eg. with or without `--privileged` ) 
- to differentiate them we can use the *name* or the *tag*
- aliasing by multiple tags that point to the same image
   - Note: we should not use multiple names for the same image, it would consume space and slow down the building process. Tags are ok for this purpose, IMHO.

**Solutions**
1. Let `falcosecurity/falco:lastest` as is now, and add `falcosecurity/falco-SOMETHING:latest` 
   - SOMETHING could be: `userspace`, `essential`, etc...
   - optionally add some aliases (eg. `:unprivileged`, `:userspace-only`, `:slim`, etc..)

2. same as 1. but using just one name  (`falcosecurity:falco`) and multiple tags

3. `falcosecurity/falco:lastest` becomes userspace only (ie. no driver, no toolchain), and add `falcosecurity/falco-SOMETHING:lastest`
   - SOMETHING could be: `full`, `equipped`, etc...
   - optionally add some aliases (eg. `:privileged`, `:with-driver-loader`, etc...)

4. same as 3. but using just one name  (`falcosecurity:falco`) and multiple tags 

5. same as 3. but using a verbose name first and using an alias later

Everyone, please pick one! :rocket:

Reply:
I go for 2. 
`falcosecurity/falco:latest` (current stable)
`falcosecurity/falco:essential` (current slim)
Aliases can be added later.

Reply:
What does everyone thinks about:

- `falcosecurity/falco` -> as it is now
- `falcosecurity/falco-no-driver` -> only the falco binary, no driver and no build toolchains (which are a dependency of the driver)

Reply:
I think we already decided that `falcosecurity/falco:latest` stay as is, for various reasons.

The issue here is about the image that today is called `slim`.

I understand the need for finding a more communicative way to name it.

We can alias/name it:
- `falcosecurity/falco-essential`
- `falcosecurity/falco-without-driver`
- `falcosecurity/falco-no-driver`

Reply:
So, it seems to me that you're agreeing on use a different image name (not just a different tag). Basically, point 1, which I can agree too.

Reply:
What about `falcosecurity/falco-driverless` ?

Reply:
I believe `falcosecurity/falco-no-driver` is the only one we can agree, so I have added it to the proposal.